inductiva/utils/autocompletion.py

```diff
@@ -0,0 +1,138 @@
+"""Utils to setup cli autocompletion."""
+from typing import List, Union, Optional
+import os
+import pathlib
+import shutil
+
+import inductiva
+from inductiva import constants
+
+BEGIN_MARKER_LINE = "# INDUCTIVA BEGIN:"
+END_MARKER_LINE = "# INDUCTIVA END:"
+
+
+def setup_zsh_autocompletion():
+    """Sets up shell auto-completion for zsh.
+
+    It does this by:
+
+    1. Adding the file with completions to:
+      `$HOME/.inductiva/v{version}/completions/_inductiva`
+    This file already comes with the `inductiva` package.
+
+    2. Adding to the users .zshrc:
+
+        # INDUCTIVA BEGIN:
+        source {setup_file_path}
+        # INDUCTIVA END:
+
+    That points to a file:
+    /.inductiva/v{version}/inductiva-setup.sh. This file than adds the
+    autocompletion script to the `fpath` of the user.
+
+    """
+    version = inductiva.__version__
+    version_dir = constants.LOCAL_LOGGING_DIR / f"v{version}"
+    completions_dir = version_dir / "completions"
+
+    os.makedirs(version_dir, exist_ok=True)
+    os.makedirs(completions_dir, exist_ok=True)
+
+    completion_file = pathlib.Path(
+        inductiva.__path__[0]) / "assets" / "completions" / "zsh" / "_inductiva"
+
+    shutil.copyfile(completion_file, completions_dir / "_inductiva")
+
+    lines_to_append = [
+        f"fpath=({completions_dir} $fpath)\n",
+        "autoload -Uz compinit\n",
+        "compinit\n",
+    ]
+
+    setup_file_path = version_dir / "inductiva-setup.sh"
+    _append_lines_to_file(lines_to_append, setup_file_path)
+
+    lines_to_append = [
+        f"\n\n{BEGIN_MARKER_LINE}\n",
+        f"source {setup_file_path}\n"
+        f"{END_MARKER_LINE}\n\n",
+    ]
+    zshrc_path = pathlib.Path.home() / ".zshrc"
+    _append_lines_to_file(lines_to_append, zshrc_path, BEGIN_MARKER_LINE,
+                          END_MARKER_LINE)
```

Comment:
this bit can be significantly simplified. The content of the bash script can be moved to the assets directory